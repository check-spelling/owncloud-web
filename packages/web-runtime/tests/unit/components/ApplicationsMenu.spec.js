import { shallowMount, createLocalVue } from '@vue/test-utils'
import DesignSystem from 'owncloud-design-system'
import GetText from 'vue-gettext'
import VueRouter from 'vue-router'

import ApplicationsMenu from 'web-runtime/src/components/ApplicationsMenu.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)
localVue.use(DesignSystem)
localVue.use(GetText, {
  translations: 'does-not-matter',
  silent: true
})

const menuLinks = [
  {
    name: 'Files',
    id: 'files',
    icon: 'folder',
    type: 'extension'
  },
  {
    name: 'External',
    id: 'external',
    icon: 'some-icon',
    type: '',
    url: 'http://some.org'
  }
]

const selectors = {
  appSwitcherButton: '#_appSwitcherButton',
  ocDrop: 'oc-drop-stub'
}

describe('ApplicationsMenu component', () => {
  jest
    .spyOn(ApplicationsMenu.mixins[0].methods, 'navigation_getMenuItems')
    .mockImplementation(() => [
      {
        iconMaterial: 'folder',
        iconUrl: undefined,
        title: 'Files',
        path: '/files'
      },
      {
        iconMaterial: 'some-icon',
        iconUrl: undefined,
        title: 'External',
        url: 'http://some.org',
        target: '_blank'
      }
    ])

  afterEach(() => {
    jest.clearAllMocks()
  })

  let wrapper
  beforeEach(() => {
    wrapper = getShallowWrapper(menuLinks)
  })

  it('should set aria-label prop on switcher button', () => {
    const appSwitcherButton = wrapper.find(selectors.appSwitcherButton)

    expect(appSwitcherButton).toMatchSnapshot()
  })
  it('should show app menus', () => {
    const ocDrop = wrapper.find(selectors.ocDrop)

    expect(ocDrop).toMatchSnapshot()
  })
})

function getShallowWrapper(applicationsList = []) {
  return shallowMount(ApplicationsMenu, {
    localVue,
    propsData: {
      applicationsList
    }
  })
}
