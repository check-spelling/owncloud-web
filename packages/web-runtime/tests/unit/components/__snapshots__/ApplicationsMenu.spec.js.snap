// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationsMenu component should set aria-label prop on switcher button 1`] = `
<oc-button-stub type="button" size="medium" arialabel="Application Switcher" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="_appSwitcherButton" class="oc-topbar-menu-burger">
  <oc-icon-stub name="apps" accessiblelabel="" type="span" size="medium" variation="passive" class="uk-flex"></oc-icon-stub>
</oc-button-stub>
`;

exports[`ApplicationsMenu component should show app menus 1`] = `
<oc-drop-stub dropid="app-switcher-dropdown" toggle="#_appSwitcherButton" position="bottom-start" mode="click" closeonclick="true" options="[object Object]" class="uk-width-large">
  <ul uk-grid="" class="uk-grid-small uk-text-center">
    <li class="uk-width-1-3">
      <router-link-stub to="/files" tag="a" ariacurrentvalue="page" event="click">
        <oc-icon-stub name="folder" accessiblelabel="" type="span" size="xlarge" variation="passive"></oc-icon-stub> <span class="uk-display-block">Files</span>
      </router-link-stub>
    </li>
    <li class="uk-width-1-3"><a target="_blank" href="http://some.org">
        <oc-icon-stub name="some-icon" accessiblelabel="" type="span" size="xlarge" variation="passive"></oc-icon-stub> <span class="uk-display-block">External</span>
      </a></li>
  </ul>
</oc-drop-stub>
`;
